tables/Tables/TableSyncData.h

```diff
@@ -120,15 +120,16 @@ class TableSyncData
     uInt        itsModifyCounter;
     uInt        itsTableChangeCounter;
     Block<uInt> itsDataManChangeCounter;
-    MemoryIO    itsMemIO;
+    MemoryIO*   itsMemIO;                  //# deleted by shared_ptr below
     AipsIO      itsAipsIO;
+    std::shared_ptr<ByteIO> itsByteIO;     //# deletes itsMemIO
```

Comment:
The shared_ptr<ByteIO> is needed by the AipsIO class. The other option is to do the somewhat complicated cast
of shared_ptr<MemoryIO> to shared_ptr<ByteIO>, which I don't particularly like.

Reply:
These casts are automatic. Since MemoryIO derives from ByteIO, you can supply a std::shared_ptr<MemoryIO> object as argument when a function expects a std::shared_ptr<ByteIO> argument.